File: src/main/java/xyz/jessyu/FetchAllIds.java
package xyz.jessyu;

import com.mongodb.ConnectionString;
import com.mongodb.client.FindIterable;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import org.bson.Document;

import java.util.ArrayList;
import java.util.List;

public class FetchAllIds {
    private static final String DB_URL = Settings.getDbUrl();
    private static final String DB_NAME = Settings.getDbName();
    private static final String DB_COLLECTION = Settings.getDbCollection();
    private static final MongoClient mongoClient = MongoClients.create(new ConnectionString(DB_URL));
    private static final MongoCollection<Document> collection = mongoClient.getDatabase(DB_NAME)
            .getCollection(DB_COLLECTION);;

    public static List<String> fetchAllIds() {

        List<String> idList = new ArrayList<>();

        // 只查 `_id` 欄位
        FindIterable<Document> docs = collection.find().projection(new Document("_id", 1));

        for (Document doc : docs) {
            idList.add(doc.get("_id").toString()); // 如果 _id 是 ObjectId 類型
        }

        return idList;
    }
}

File: src/python/xyz/jessyu/fetch_all_ids.py
"""
MongoDB operations for fetching existing post IDs.
"""
from pymongo import MongoClient
from typing import List
from .settings import Settings


def fetch_all_ids() -> List[str]:
    """
    Fetch all post IDs from MongoDB database.
    
    Returns:
        List of post ID strings
    """
    try:
        db_url = Settings.get_db_url()
        db_name = Settings.get_db_name()
        db_collection = Settings.get_db_collection()
        
        if not db_url:
            return []
        
        client = MongoClient(db_url)
        collection = client[db_name][db_collection]
        
        # Query only the _id field
        docs = collection.find({}, {"_id": 1})
        
        id_list = [str(doc["_id"]) for doc in docs]
        
        client.close()
        return id_list
    except Exception as e:
        print(f"Error fetching IDs from database: {e}")
        return []
